packages/next/pages/_document.js

```diff
@@ -50,7 +50,11 @@ export class Html extends Component {
 
   render() {
     const { amphtml } = this.context._documentProps
-    return <html amp={amphtml ? '' : null}>{this.props.children}</html>
+    return (
+      <html amp={amphtml ? '' : null} {...this.props}>
```

Comment:
maybe for consistency the `amp` attribute should always come from context and override `props.amp`. Probably the amp validator will complain about this but you could spare that to the end user

Reply:
Yeah, I guess if `amphtml` is set we are rendering as AMP -- nothing the user can do to opt out.